src/main/java/io/wispforest/owo/registration/reflect/BlockRegistryContainer.java

```diff
@@ -14,41 +14,48 @@
 import java.lang.annotation.Target;
 import java.lang.reflect.Field;
 
-public interface BlockRegistryContainer extends AutoRegistryContainer<Block> {
-
-    @Override
-    default Registry<Block> getRegistry() {
-        return Registries.BLOCK;
-    }
-
-    @Override
-    default Class<Block> getTargetFieldType() {
-        return Block.class;
-    }
-
-    @Override
-    default void postProcessField(String namespace, Block value, String identifier, Field field) {
-        if (field.isAnnotationPresent(NoBlockItem.class)) return;
-        Registry.register(Registries.ITEM, Identifier.of(namespace, identifier), createBlockItem(value, identifier));
-    }
-
-    /**
-     * Creates a block item for the given block
-     *
-     * @param block      The block to create an item for
-     * @param identifier The identifier the field was assigned, possibly overridden by an {@link AssignedName}
-     *                   annotation and always fully lowercase
-     * @return The created BlockItem instance
-     */
-    default BlockItem createBlockItem(Block block, String identifier) {
-        return new BlockItem(block, new Item.Settings());
-    }
-
-    /**
-     * Declares that the annotated field should not
-     * have a block item created for it
-     */
-    @Retention(RetentionPolicy.RUNTIME)
-    @Target(ElementType.FIELD)
-    @interface NoBlockItem {}
+/**
+ * Due to changes within 1.21.2 makes {@link AutoRegistryContainer} impossible due to requiring
+ * the {@link net.minecraft.registry.RegistryKey} leading to the need to supplier the key before
+ * fully creating the desired entry.
+ * </br></br>
+ * Recommend using Minecraft methods for registering such entry
+ */
+@Deprecated(forRemoval = true)
+public interface BlockRegistryContainer /*extends AutoRegistryContainer<Block>*/ {
```

Comment:
if you're removing all methods and making the interface basically useless why not just remove it?
in any case imo this and the item registry container should be able to be saved if they're changed to work more like in the field registration adjustments pr

Reply:
Done such as a way to illustrate why it was removed, can be removed if such is not desired but find this more helpful than just removing outright